### PR TITLE
rename to nnp 

### PR DESCRIPTION
**Pull request by sasadep**
this PR is dependent https://github.com/NervanaSystems/argon-transformer/pull/167
the `ARGON` manager is changes to `NNP` manager along with file names and tests

### CODE REVIEW

File: cmake/external_nnp_transformer.cmake [hunk 1]
```diff
@@ -16,133 +16,133 @@
 
 # To build ngraph with Argon transformer using pre-build Argon API
 # ```
-# cmake -DNGRAPH_ARGON_ENABLE=True -DNGRAPH_PREBUILD_ARGON_API_PATH=$HOME/dev/system/_out/debug-x86_64-Linux ..
+# cmake -DNGRAPH_NNP_ENABLE=True -DNGRAPH_PREBUILD_NNP_API_PATH=$HOME/dev/system/_out/debug-x86_64-Linux ..
```
Comment (yxlao) [inline cmake/external_nnp_transformer.cmake]: `DNGRAPH_PREBUILD_ARGON_API_PATH` since it is still `ARGON_API`

File: cmake/external_nnp_transformer.cmake [hunk 2]
```diff
@@ -16,38 +16,38 @@
...
+    # - Before we have ngraph CI job for nnp transformer, ngraph master might not be
+    #   compatible with nnp transformer. To ensure compatibility, checkout the ngraph commit point
+    #   where the following `NNP_TRANSFORMER_GIT_TAG` is set and build ngraph with nnp using this
     #   commit.
-    # - After we have ngraph CI job for argon transformer, ngraph master will be compatible with
-    #   argon transformer guaranteed by CI.
-    set(ARGON_TRANSFORMER_GIT_TAG cpp-master)
+    # - After we have ngraph CI job for nnp transformer, ngraph master will be compatible with
+    #   nnp transformer guaranteed by CI.
+    set(NNP_TRANSFORMER_GIT_TAG sandeep/rename-nnp)
```
Comment (yxlao) [inline cmake/external_nnp_transformer.cmake]: We shall keep it as `cpp-master`, and `git checkout` locally if we want to test other branches.

File: python/pyngraph/ops/batch_norm.cpp [hunk 3]
```diff
@@ -41,7 +41,8 @@ void regclass_pyngraph_op_BatchNormBackprop(py::module m)
                std::shared_ptr<ngraph::op::BatchNormBackprop>,
                ngraph::op::util::RequiresTensorViewArgs>
         batch_norm_backprop(m, "BatchNormBackprop");
-    batch_norm_backprop.doc() = "ngraph.impl.op.BatchNormBackprop wraps ngraph::op::BatchNormBackprop";
+    batch_norm_backprop.doc() =
```
Comment (yxlao) [inline python/pyngraph/ops/batch_norm.cpp]: Does our `clang-format` checks the `python` directory as well?

File: test/nnp_fusion.cpp [hunk 4]
```diff
@@ -43,7 +43,7 @@
 using namespace ngraph;
 using namespace std;
 
-TEST(Argon_fusion, fuse_max_with_constant_zero_input_as_relu)
+TEST(nnp_fusion, fuse_max_with_constant_zero_input_as_relu)
```
Comment (yxlao) [inline test/nnp_fusion.cpp]: Should this be moved to the nnp repo?

File: python/pyngraph/ops/batch_norm.cpp (hunk 3 above)
Comment (sasadep) [reply python/pyngraph/ops/batch_norm.cpp]: yup! It does [LINK](https://github.com/NervanaSystems/ngraph/blob/master/maint/apply-code-format.sh#L47)

File: test/nnp_fusion.cpp (hunk 4 above)
Comment (sasadep) [reply test/nnp_fusion.cpp]: I need to move this to backendtest

Comment (sasadep) [reply test/nnp_fusion.cpp]: removed the tests as it was taken care in the `CoreFusion` PR

File: cmake/external_nnp_transformer.cmake [hunk 5]
```diff
@@ -16,38 +16,38 @@
...
 
-    # Determines where argon-transformer will be located
-    set(ARGON_TRANSFORMER_PREFIX ${CMAKE_CURRENT_BINARY_DIR}/argon_transformer)
+    # Determines where nnp-transformer will be located
+    set(NNP_TRANSFORMER_PREFIX ${CMAKE_CURRENT_BINARY_DIR}/argon_transformer)
```
Comment (yxlao) [inline cmake/external_nnp_transformer.cmake]: `${CMAKE_CURRENT_BINARY_DIR}/nnp_transformer`

File: maint/apply-code-format.sh [hunk 6]
```diff
@@ -42,12 +42,12 @@ bash_lib_status "Verified that '${CLANG_FORMAT_PROG}' has version '${REQUIRED_CL
 
 pushd "${THIS_SCRIPT_DIR}/.."
 
-declare ARGON_SRC_DIR="build/third-party/argon_transformer/src/ext_argon_transformer/src"
-declare ARGON_TEST_DIR="build/third-party/argon_transformer/src/ext_argon_transformer/test"
+declare NNP_SRC_DIR="build/third-party/argon_transformer/src/ext_nnp_transformer/src"
```
Comment (yxlao) [inline maint/apply-code-format.sh]: `third-party/nnp_transformer`

File: maint/check-code-format.sh [hunk 7]
```diff
@@ -45,11 +45,11 @@ declare NUM_FILES_CHECKED=0
 
 pushd "${THIS_SCRIPT_DIR}/.."
 
-declare ARGON_SRC_DIR="build/third-party/argon_transformer/src/ext_argon_transformer/src"
-declare ARGON_TEST_DIR="build/third-party/argon_transformer/src/ext_argon_transformer/test"
+declare NNP_SRC_DIR="build/third-party/argon_transformer/src/ext_nnp_transformer/src"
+declare NNP_TEST_DIR="build/third-party/argon_transformer/src/ext_nnp_transformer/test"
```
Comment (yxlao) [inline maint/check-code-format.sh]: `third-party/nnp_transformer`

**Comment by yxlao** [comment]
verified building both PR
